src/main/java/de/hpi/datastreams/processors/PredictionProcessor.java

```diff
@@ -0,0 +1,40 @@
+package de.hpi.datastreams.processors;
+
+import de.hpi.datastreams.apps.App;
+import de.hpi.datastreams.messages.DataMessage;
+import de.hpi.datastreams.messages.MyHashMap;
+import de.hpi.datastreams.messages.MySimpleHashMap;
+import de.hpi.datastreams.producer.ProducerBuilder;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.streams.processor.AbstractProcessor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.state.KeyValueStore;
+
+import java.util.UUID;
+
+import static de.hpi.datastreams.apps.App.PREDICTION_OUTPUT_TOPIC;
+
+public class PredictionProcessor extends AbstractProcessor<Long, DataMessage> {
+
+    private KeyValueStore<Long, MySimpleHashMap> weights;
+    private Producer<Long, Float> predictionMessageProducer;
+
+    @Override
+    public void init(ProcessorContext context) {
+        super.init(context);
+        this.weights = (KeyValueStore<Long, MySimpleHashMap>) context.getStateStore(App.WEIGHTS_STORE);
```

Comment:
Partitioning the weights across the cluster is probably not what we should do here.
We could either use a global store or rely on the weight stores of the training processors.
I would prefer the latter because this allows us to keep the number of required messages quite low as the `TrainingProcessor`s anyway have to get weight updates.
The global store would introduce additional traffic as all weight updates have to be broadcasted to all other partitions than the one the `ServerProcessor` itself is executed on.

Reply:
Why would you want to rely on the weights of the training processors? The Server has the most complete view of the weights because he is the one updating the weight (while the training processors pull the weight from it). 

In the current implementation, the PredictionProcessor accesses the weight from the Server.
As far as I understand it, the KeyValueStore is partitioned across the nodes by the key. 
We said that we only want one Server and one PredictionProcessor.  Therefore, by always setting the key to '0' means that all data is stored on a single node. 
As long as we have only one server and one PredictionsProcessor which are both running on node 0, this should work well. (Although, I am not sure yet how we can make sure that the Server and the PredictionProcessor always run on the node 0)

If we were using a GlobalStore, then the weights would need to be shared across all nodes (I think this would be overkill).

I am not sure if I understood you correctly. Why do you think that we should rely on the training processors?